public/app/assetlibrary/assetlibrary_list/assetLibraryListController.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2015 UC Berkeley (UCB) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+(function(angular) {
+
+  'use strict';
+
+  angular.module('collabosphere').controller('AssetLibraryListController', function(assetLibraryListFactory, userService, utilService, $location, $scope) {
+
+    $scope.assets = [];
+    $scope.list = {
+      'page': 0,
+      'isLoading': false
+    };
+
+    /**
+     * Get the assets for the current course through an infinite scroll
+     */
+    var getAssets = $scope.getAssets = function() {
+      $scope.list.isLoading = true;
+      assetLibraryListFactory.getAssets($scope.list.page).success(function(assets) {
+        $scope.assets = $scope.assets.concat(assets.results);
+        if (assets.results.length === 10) {
```

Comment:
Actually, there wasn't too much clean-up work I could necessarily do here, so I've added some comments to this block.